src/rust/types/src/events.rs

```diff
@@ -41,22 +42,80 @@ pub enum CancelReason {
     ViolatedLimitPrice,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(rename_all = "snake_case")
+)]
+pub enum MakerEventType {
+    Cancel,
+    Change,
+    Evict,
+    Place,
+}
+
+impl TryFrom<u8> for MakerEventType {
+    type Error = TypeError;
+
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(MakerEventType::Cancel),
+            1 => Ok(MakerEventType::Change),
+            2 => Ok(MakerEventType::Evict),
+            3 => Ok(MakerEventType::Place),
+            _ => Err(TypeError::ConversionError {
+                name: "MakerEventType".to_string(),
+            }),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct MakerEvent {
+    pub market_id: u64,
+    pub side: Side,
+    pub market_order_id: u128,
+    pub user_address: String,
+    pub custodian_id: Option<u64>,
+    pub event_type: MakerEventType,
+    pub size: u64,
+    pub price: u64,
+    pub time: DateTime<Utc>,
+}
+
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct TakerEvent {
+    pub market_id: u64,
+    pub side: Side,
+    pub market_order_id: u128,
+    pub maker: String,
+    pub custodian_id: Option<u64>,
+    pub size: u64,
+    pub price: u64,
+}
+
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct Type {
```

Comment:
How about switching this to `TypeInfo` to be more consistent with the Move type?